config/initializers/diaspora_federation.rb

```diff
@@ -61,5 +35,26 @@
 
       person_entity.save!
     end
+
+    on :fetch_private_key_by_id do |diaspora_id|
+      Person.find_by(diaspora_handle: diaspora_id).owner.encryption_key
+    end
+
+    on :fetch_private_key_by_post_guid do |guid|
+      author_user = Post.find_by(guid: guid).author.owner
+      author_user.encryption_key unless author_user.nil?
+    end
+
+    on :fetch_public_key_by_id do |diaspora_id|
+      Person.find_by(diaspora_handle: diaspora_id).author.public_key
+    end
+
+    on :fetch_public_key_by_post_guid do |guid|
+      Post.find_by(guid: guid).author.public_key
+    end
+
+    on :post_author_is_local? do |guid|
+      Post.find_by(guid: guid).author.local?
```

Comment:
Well, yes, we know only parent post guid. Look at the [comment structure](https://github.com/SuperTux88/diaspora_federation/blob/salmon/lib/diaspora_federation/entities/relayable.rb). Instead I can propose to add a hook ```fetch_diaspora_id_by_post_guid``` then and call two hooks consequently to receive information about parent author.